pandas/core/indexes/category.py

```diff
@@ -512,10 +512,8 @@ def _reindex_non_unique(self, target):
     # --------------------------------------------------------------------
     # Indexing Methods
 
-    def _maybe_cast_indexer(self, key):
-        code = self.categories.get_loc(key)
-        code = self.codes.dtype.type(code)
-        return code
+    def _maybe_cast_indexer(self, key) -> int:
+        return self._data._unbox_scalar(key)
```

Comment:
would it make sense to align the method names here

Reply:
im aiming to get there before long, but not there yet.  Index and subclasses still have several similar-but-not-identical methods like this